### PR TITLE
ICMSLST-2282 Add view to recover ICMS V1 account.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Changes:
  - Add view to allow a user to recover their V1 account
  - Update ICMSStaffSSOBackend to check User.is_active
  - Add two fake v1 migrated users to test account recovery view.
  - Add tests for backend, utils, forms and view.

### CODE REVIEW

**Comment by MattHolmes123** [comment]
The best way to test the migration is to do the following:

1. Log in as one of the new legacy users (migrated-user-1@example.com, migrated-user-2@example.com)
2. Create an application
3. login as a v2 user (e.g. importer_user)
4. Navigate to new screen (Click name on top menu > My Details > Account Recovery)
5. Fill form out using legacy user login details)
6. log in again as v2 user to see data migrated from legacy user.

**Review by mikeylockwood — APPROVED** [review]
I guess one potential issue with this could be if before recovery, the newly created user is added to an importer and exporter and raises an application.  Would these applications still be editable by the new user? How could this be prevented?

File: web/auth/backends.py [hunk 1]
```diff
@@ -141,6 +150,16 @@ def get_or_create_user(self, profile: StaffSSOProfile) -> User:
 
         return user
 
+    def user_can_authenticate(self, user: User) -> bool:
+        """Reject users with is_active=False.
+
+        Custom user models that don't have that attribute are allowed.
+        """
+
+        is_active = getattr(user, "is_active", None)
```
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/auth/backends.py:159]: I just straight up copied the ModelBackend code.

**Comment by MattHolmes123** [comment]
> I guess one potential issue with this could be if before recovery, the newly created user is added to an importer and exporter and raises an application. Would these applications still be editable by the new user? How could this be prevented?

No there is a warning on the page to say existing data will be lost

**Review by marcuspp — APPROVED** [review]

File: web/auth/backends.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/auth/backends.py:159]: Couldn't think of a good reason to deviate away from what django did so left it as is.

**Pull request merged**